DependencyInjection/Compiler/RegisterFastestCachePass.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\DoctrineBundle\CacheWarmer\DoctrineMetadataCacheWarmer;
+use Symfony\Component\Cache\Adapter\PhpArrayAdapter;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+use function array_keys;
+use function preg_replace;
+use function sprintf;
+
+/** @internal */
+final class RegisterFastestCachePass implements CompilerPassInterface
```

Comment:
I suppose this pass is used to wrap the configured metadata cache in the PhpArrayAdapter? If so, I'd propose a simpler alternative:
* If no metadata cache is configured, configure an `ArrayAdapter` or `PhpArrayAdapter` depending on whether we're in debug mode
* If a metadata cache is configured use only that configuration.

Users will see a deprecation notice when configuring a metadata cache and will be inclined to remove that configuration; at that point they'll automatically get a faster cache. I don't think we need to wrap their configured metadata cache at all and reduce the amount of logic needed here.

Reply:
PhpArrayAdapter requires PSR6 cache as an argument. But turning doctrine cache into PSR6 cache is done in compiler pass, that's why I had to remove creation of PhpArrayAdapter and put it to another compiler pass which is executed after PSR-6 cache has been createdin first compiler pass. But feel free to experiment with this and move it back to DoctrineExtension if you find some way which makes all tests to pass.

Reply:
I've removed this pass and simplified the compatibility layer.

We now only register a PhpArrayAdapter if no cache is configured - this is to preserve original behaviour where people want to have their specific cache for a reason. If a cache is configured, there is no change to that apart from making sure it implements PSR-6 (either directly or through an adapter). Given that people should drop the `metadata_cache_driver` configuration entirely in favour of the automatic cache handling I don't think there's any value in having that extra logic.